flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EventHelper.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.operator.utils;
+
+import org.apache.flink.kubernetes.operator.crd.AbstractFlinkResource;
+import org.apache.flink.kubernetes.operator.crd.FlinkDeployment;
+import org.apache.flink.kubernetes.operator.listener.FlinkResourceListener;
+
+import io.fabric8.kubernetes.api.model.Event;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.function.BiConsumer;
+
+/** Helper class for status management and updates. */
+public class EventHelper {
```

Comment:
`EventHelper` is a bit awkward from a naming perspective, since this is a pluggable component that needs to be injected everywhere. I see that you want to separate the creation of the event from the listener. Would `EventEmitter` be a fit for this?

Reply:
I am a bit torn here, this class doesn't emit events on its own. So maybe `EventUtil` would be better if we dont like the `helper` name. We also have `StatusHelper` now that could also be renamed to `StatusUtil`. These classes create an interface between the operator and the kubernetes client so we can control these interactions better.

Reply:
Maybe we can look at it from the client perspective rather than how it is implemented? It functions as a kind of "sink" for events.

Reply:
It is called EventRecorder in kubebuilder:
https://book-v1.book.kubebuilder.io/beyond_basics/creating_events.html

Reply:
We could probably request some JOSDK API support for this feature. What do you think @csviri?

Reply:
`EventRecorder` and `StatusRecorder` sounds perfect, thanks @morhidi , will make this change

Reply:
`EventReconder` makes sense for sure, I think `StatusReconder` we discussed before that it should be rather an other resource where the state is recorded (in general). 
But please create an issue that describes the uses cases and/or motivation for JOSDK! Eventually EventRecorder will be supported for sure. We can discuss the other. 

If you think the `FlinkResourceListener` makes sense to generalize, please describe the motivation behind that too.